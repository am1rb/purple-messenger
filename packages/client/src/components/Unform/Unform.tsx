import React, { useMemo } from "react";
import { Form, SubmitHandler } from "@rocketseat/unform";
import { FormProps } from "@rocketseat/unform/dist/Form";

interface UnformProps<T> extends Omit<FormProps, "initialData" | "onSubmit"> {
  initialData?: Partial<T>;
  onSubmit: SubmitHandler<T>;
}

function Unform<T>({ initialData: initialDataProp, ...other }: UnformProps<T>) {
  const initialData = useMemo(() => initialDataProp, []);
  return <Form {...(other as FormProps)} initialData={initialData} />;
}
export default Unform;
