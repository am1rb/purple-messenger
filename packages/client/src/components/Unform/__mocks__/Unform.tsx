import React from "react";
import { SubmitHandler } from "@rocketseat/unform";
import { StringObject } from "core/type/general";

interface UnformProps<T> {
  initialData?: Partial<T>;
  onSubmit: SubmitHandler<T>;
  children: React.ReactNode;
}

const helpers = {
  resetForm: () => {},
};

function Unform<T>({ initialData, onSubmit, children }: UnformProps<T>) {
  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
    event.preventDefault();

    const data: StringObject = {};

    const elements = event.currentTarget.elements;
    for (let i = 0; i < elements.length; i++) {
      const input = elements[i] as HTMLInputElement;
      data[input.name] = input.value;
    }

    onSubmit(data as T, helpers as any);
  }

  return <form onSubmit={handleSubmit}>{children}</form>;
}
export default Unform;
