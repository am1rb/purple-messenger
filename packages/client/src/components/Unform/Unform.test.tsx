import React from "react";
import { render } from "@testing-library/react";
import Unform, { UnformProps } from "./Unform";

const sharedProps: UnformProps<unknown> = {
  onSubmit: () => {},
  children: <>Form Fields</>,
};

describe("The <Unform /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(<Unform {...sharedProps} />);
    expect(container).toMatchSnapshot();
  });
});
