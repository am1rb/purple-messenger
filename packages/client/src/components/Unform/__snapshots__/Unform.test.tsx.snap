// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Unform /> tests Should match the snapshot 1`] = `
<div>
  <form
    data-testid="mock-form"
  >
    Form Fields
  </form>
</div>
`;
